frontend/src/helpers/Analytics.ts

```diff
@@ -0,0 +1,86 @@
+import { version } from '../../package.json'
+
+/// <reference types="@types/segment-analytics" />
+///  <reference types="@types/google.analytics" />
+
+export default class Analytics {
+  private static _instance: Analytics
+  private context: any
+  private gaAppSet: boolean
+
+  private constructor() {
+    this.gaAppSet = false
+    this.context = {
+      category: 'Desktop',
+      productName: 'Desktop',
+      productVersion: version,
+      //retrieved from backend
+      systemOS: '',
+      systemOSVersion: '',
+      systemArch: '',
+      manufacturerName: '',
+      manufacturerProductVersion: '',
+      manufacturerProductName: '',
+      manufacturerProductCode: '',
+      manufacturerPlatformName: '',
+      manufacturerPlatformCode: '',
+    }
+  }
+
+  public setManufacturerDetails(details: IManufacturer) {
+    this.context.manufacturerName = details.name
+    this.context.manufacturerProductVersion = details.product.version
+    this.context.manufacturerProductName = details.product.name
+    this.context.manufacturerProductCode = details.product.code
+    this.context.manufacturerPlatformName = details.platform.name
+    this.context.manufacturerPlatformCode = details.platform.code
+  }
+
+  public setArch(arch: any) {
+    this.context.systemArch = arch
+  }
+
+  public setOS(os: any) {
+    this.context.systemOS = os
+  }
+
+  public setOsVersion(version: any) {
+    this.context.systemOSVersion = version
+  }
+
+  public static get Instance() {
```

Comment:
This is kinda weird to me. What about just create an instance and export that instead, like:

```js
class Analytics {}

export default new Analytics()
```

Then make all methods not static.

Reply:
Yes please.

Reply:
I could be wrong but to me that seems like that would create a new Analytics object every time you imported it.

Reply:
No, this is the standard way to create a Singleton instance of a class. It will only be executed once.

Reply:
updated with this method but it did cause some issues with initialization.  now must call window.analytics directly from inside Analytics class rather then using an object variable.  Jamie saw this issue with me.